apps/emqx/src/emqx_hooks.erl

```diff
@@ -140,7 +140,7 @@ add(HookPoint, Action, Priority) when is_integer(Priority) ->
 add(HookPoint, Action, Filter, Priority) when is_integer(Priority) ->
     add(HookPoint, #callback{action = Action, filter = Filter, priority = Priority}).
 
-%% @doc Like add/2, it register a callback, discard 'already_exists' error.
+%% @doc `put/2,3,4` updates the existing hook, add it if not exists.
```

Comment:
should we continue to provide the apis without priority ?
maybe remove to enforce awareness .

Reply:
Some of the user may want to control the priority of a specific app, mainly for the `exhook`. 

The exhook can be configured to start with the highest priority, so that the user can write his own code to stop the message from being handled by the rule engine.

Reply:
Done